docs/resource_sets.rst

```diff
@@ -0,0 +1,93 @@
+
+
+Constraints and rules
+************************
+
+Some restrictions and guidelines were implemented to guarantee that partial compilations function properly:
+
+* A resource cannot be a part of more than one resource set at once.
+* A resource does not have to be part of a resource set.
+* Resources cannot be migrated using a partial compile to a different resource set. A full compile is necessary for this process.
+* A resource set that is contained in a partial export must be complete, meaning that all of its resources must be present.
+* Resources that weren't assigned to a specific resource set can never be updated or removed by a partial build. Although, adding resources is allowed.
+* The new version of the model that emerges from a partial compilation should have a dependency graph that is closed within the resource sets that were exported.
+  i.e., it should not depend on any resource sets other than those that were exported.
+* Multiple resource sets may be updated simultaneously via a partial build.
+
+
+Exporting a partial model to the server
+******************************************************
+
+Two arguments can be passed to the ``inmanta export`` command in order to export a partial model to the server:
+
+- ``--partial`` To specify that the model being compiled only contains the resources that need to be updated in relation to the previous version of the model.
+- ``--delete-resource-set <resource-set-name>`` This option, which may be used more than once, instructs the model to remove the resource set with the specified name. Only in conjunction with the preceding choice may this option be utilized. Note that utilizing a ``std::ResourceSet`` that includes no resources allows resource sets to be implicitly deleted during a partial compilation.
+
+
+Limitations
+*************
+
+* The compiler cannot verify all constraints that would be verified when a full build is run. Some index constraints, for instance, cannot be verified. The model creator is in charge of making sure that these constraints are met.
+* If just a partial compile is performed, it is possible for a shared resource to become obsolete because numerous resources from distinct resource sets may depend on the same shared resource (not associated with a specific resource set).
+  The shared resource will become obsolete when a partial compile deletes the last resource that depended on it, but it is preserved as a server-managed resource because partial compiles cannot delete shared resources.
+  A full compile is required to remove shared resources. Scheduled complete compilations that ``garbage-collect`` these shared resources are one way to fix this.
+  The :inmanta.environment-settings:setting:`auto_full_compile` environment setting is used to schedule full compilations.
+  As an example, to plan a daily full compile for 01:00 UTC, use the ``auto full compile`` environment setting:  "0 1 * * *".
```

Comment:
I don't like the formatting of the cron expression but it don't know what the conventions are so maybe no change is needed.

Reply:
1. It's already been released so there's little we can do at this point.
2. It really just follows the cron format so while it's not the most intuitive it is familiar to many.

Reply:
I meant the how it is displayed as 'normal' text in the doc, the layout. I don't have an issue with the cron format in itself :p

Reply:
Ah, I see, that we did change because Arnaud requested the same.